src/box/lua/config/instance_config.lua

```diff
@@ -1513,6 +1513,139 @@ return schema.new('instance_config', schema.record({
             value = schema.scalar({type = 'string'}),
         }),
     }),
+    sharding = schema.record({
+        -- Instance vshard options.
+        roles = schema.set({
+            'router',
+            'storage',
+        }),
```

Comment:
Shouln't we allow it on the replicaset level, but not on the instance level?

Reply:
I don't insist on prohibiting right now, but I would mark the options as related to a replicaset (even if they're currently accepted for an instance).

Reply:
As I see, it is placed into the replicaset options now without actual validation that it is not set in the instance scope. It is OK for now. Thanks!